File: ABC/299/C_Dango.py
N = int(input())
S = input()
ans = 0
temp = 0
for i in range(N):
    if S[i] == 'o':
        temp += 1
    else:
        ans = max(ans, temp)
        temp = 0

temp = 0
for i in range(N):
    if S[N-i-1] == 'o':
        temp += 1
    else:
        ans = max(ans, temp)
        temp = 0

if ans == 0:
    print(-1)
else:
    print(ans)

File: ABC/250~299/299/C_Dango.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
int main(){
    int N; cin >> N;
    string S; cin >> S;
    int ans = 0, temp = 0;
    for (int i = 0; i < N; i++) {
        if (S[i] == 'o') {
            temp++;
        }
        else {
            ans = max(ans, temp);
            temp = 0;
        }
    }
    temp = 0;
    for (int i = 0; i < N; i++) {
        if (S[N-i-1] == 'o') {
            temp++;
        }
        else {
            ans = max(ans, temp);
            temp = 0;
        }
    }
    ans == 0 ? cout << -1 << endl : cout << ans << endl;
    return 0;
}